### PR TITLE
Adding RBAC Mintable token

### PR DESCRIPTION
**Pull request by vittominacori**
# 🚀 Description

This add a RBAC Mintable token that makes addresses with the "minter" role able to mint tokens.
This could help minting in multi phases Crowdsales or, for instance, if owner wants to deploy multiple smart contracts simultaneously for doing different works (like Airdrop, Bounty, ICOs...).

- [x] 📘 I've reviewed the [OpenZeppelin Contributor Guidelines](../blob/master/CONTRIBUTING.md)
- [x] ✅ I've added tests where applicable to test my new functionality.
- [x] 📖 I've made sure that my contracts are well-documented.
- [x] 🎨 I've run the JS/Solidity linters and fixed any issues (`npm run lint:all:fix`).

### CODE REVIEW

**Comment by shrugs** [comment]
love it!

can we share more code with the existing `MintableToken` by moving the shared parts to a `Mintable` contract or something? If not, no worries. Likewise for the tests; it'd be good to have a `Mintable.behvaior.js` that shares the same check but differs in the logic used to authenticate the minter.

**Comment by vittominacori** [comment]
Yes @shrugs. 
I can share mintable functions and make a behaviour for tests.
I haven't done because of I didn't want to change Mintable token but I can update asap.
Thank you.

**Comment by shrugs** [comment]
😍 amazing!!

My only concern is that `RBACWithAdmin` may or may not be a security concern; an `admin` account is actually really powerful, since it has the `adminAddRole` and `adminRemoveRole`; it allows an admin of a contract to control every part of the access profile. We also already have an `owner` since it inherits from `Ownable` though `MintableToken`.

I'm worried that anything that implement RBACMintableToken will actually end up accidentally allowing an admin to change any role in the system without realizing it. It's also confusing to have both an owner and an admin in one contract.

I think what would be best is if the `RBACMintableToken` contract just declared something like

```
contract RBACMintableToken is MintableToken, RBAC {
  /**
   * A constant role name for indicating minters.
   */
  string public constant ROLE_MINTER = "minter";
  string public constant ROLE_OWNER = "finish_minting";

  modifier hasMintPermission() {
    checkRole(msg.sender, ROLE_MINTER);
    _;
  }

  modifier hasFinishMintingPermission() {
     require(msg.sender == owner);
    _;
  }

  function addMinter (address _addr) onlyOwner { ... }
  function removeMinter (address _addr) onlyOwner { ... }
}

```

so that the abilities of the owner are strictly defined and you can add more in the future. I'll update `RBACWithAdmin` to add a warning about this now.

**Comment by vittominacori** [comment]
OK right, will inherit from RBAC and add only the specific function to add the minter role.

I have only a note about your code.
This:

`modifier hasFinishMintingPermission() {
     require(msg.sender == owner);
    _;
  }`

is equal to the onlyOwner modifier so, instead of adding this, I will back the finishMinting permission in `MintableToken` to the onlyOwner modifier and remove the hasFinishMinting permission. 
By doing so, the ROLE_OWNER role is unused.

**Comment by vittominacori** [comment]
@shrugs done!
Also created a minted crowdsale behaviour to be tested with both mintable and rbac-mintable token.

File: test/token/ERC20/MintableToken.behaviour.js [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+const BigNumber = web3.BigNumber;
+
+require('chai')
+  .use(require('chai-as-promised'))
+  .use(require('chai-bignumber')(BigNumber))
+  .should();
+
+export default function ([owner, anotherAccount, minter]) {
+  describe('as a basic mintable token', function () {
```
Comment (shrugs) [inline test/token/ERC20/MintableToken.behaviour.js]: I'm going to merge these changes, but in the future, there's a `context` helper from Mocha for this feature; it's the same thing as describe, but helps the reader separate a "context" from the description of what the thing should do

**Pull request merged**